From f8fd257fc44af1758cfd674a0aedec17e91fbe7d Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Thu, 8 Mar 2018 18:06:40 -0800
Subject: [PATCH 1/3] Use BUILD_SHARED_LIBS

The nghttp2 library was always configured to compile as SHARED. This makes
it so BUILD_SHARED_LIBS is used to determine how nghttp2 is created.
---
 lib/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 0846d067..f69254f3 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -38,12 +38,15 @@ if(WIN32)
 endif()
 
 # Public shared library
-add_library(nghttp2 SHARED ${NGHTTP2_SOURCES} ${NGHTTP2_RES})
+add_library(nghttp2 ${NGHTTP2_SOURCES} ${NGHTTP2_RES})
 set_target_properties(nghttp2 PROPERTIES
   COMPILE_FLAGS "${WARNCFLAGS}"
   VERSION ${LT_VERSION} SOVERSION ${LT_SOVERSION}
   C_VISIBILITY_PRESET hidden
 )
+if (NOT BUILD_SHARED_LIBS)
+  target_compile_definitions(nghttp2 PUBLIC "-DNGHTTP2_STATICLIB")
+endif()
 target_include_directories(nghttp2 INTERFACE
     "${CMAKE_CURRENT_BINARY_DIR}/includes"
     "${CMAKE_CURRENT_SOURCE_DIR}/includes"
-- 
2.14.1.windows.1

